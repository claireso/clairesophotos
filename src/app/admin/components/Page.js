import React from 'react'
import PropTypes from 'prop-types'

const Page = (props = {}) => {
  return (
    <main>
      <h1>{props.title}</h1>
      {props.children}
    </main>
  )
}

Page.propTypes = {
  title: PropTypes.string.isRequired,
  children: PropTypes.node
}

export default Page
