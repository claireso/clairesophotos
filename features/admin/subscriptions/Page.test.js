import { render, waitFor } from '@testing-library/react'

import withTestRouter from '@utils/hoc/withTestRouter'

import Page from '../../../pages/admin/subscriptions/index'

describe('Page subscriptions', () => {
  const renderComponent = (routerProps) => {
    return render(
      withTestRouter(<Page />, {
        pathname: '/admin/subscriptions',
        ...routerProps
      })
    )
  }

  beforeEach(() => {
    fetch.resetMocks()
    fetch.mockResponseOnce(JSON.stringify(global.__SUBSCRIPTIONS__))
  })

  test('should load and render subscriptions', async () => {
    const { container, getByText } = renderComponent()

    expect(container).toMatchSnapshot()

    await waitFor(() => {
      expect(getByText('https://fcm.googleapis.com/')).toBeInTheDocument()
    })

    expect(container).toMatchSnapshot()
  })

  test('should display modal delete', async () => {
    const { getByText } = renderComponent({
      query: {
        action: 'delete_subscription',
        id: 199
      }
    })

    await waitFor(() => {
      expect(getByText('Are you sure?')).toBeInTheDocument()
    })
  })
})
