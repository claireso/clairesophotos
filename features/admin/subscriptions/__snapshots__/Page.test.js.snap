// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page subscriptions should load and render subscriptions 1`] = `
.c2 {
  margin: 2rem 0;
  -webkit-animation: jBcSpD 800ms ease-in-out infinite alternate;
  animation: jBcSpD 800ms ease-in-out infinite alternate;
}

.c2:after {
  background: var(--primary);
  border-radius: 50%;
  content: '';
  display: block;
  height: 1rem;
  width: 1rem;
  margin: 0 auto;
}

.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  margin: 0 0 2rem;
}

.c0 > h1 {
  line-height: 1.835;
  margin: 0;
}

.c0 > h1 > span {
  color: var(--gray-2);
  font-size: 2rem;
  font-weight: normal;
  line-height: 1;
}

.c1 {
  font-size: 2.4rem;
  margin: 0 0 2.4rem;
}

<div>
  <div
    class="c0"
  >
    <h1
      class="c1"
    >
      Your subscriptions 
      <span>
        (
        )
      </span>
    </h1>
  </div>
  <div
    class="c2"
  />
</div>
`;
